### PR TITLE
Added documentation (and enforcement) for yargs

### PR DESCRIPTION
**Pull request by UncleSamSwiss**
**PR Checklist:**

-   [x] Provide a meaningful description to this PR or mention which issues this fixes.
-   [x] Add tests for your change. This includes negative tests (i.e. inputs that need to fail) as well as baseline tests (i.e. how should the directory structure look like?). _not necessary_
-   [x] Run the test suite with `npm test`
-   [x] If there are baseline changes, review them and make a separate commit for them with the comment "accept baselines" if they are desired changes _not necessary_
-   [x] Ensure the project builds with `npm run build`
-   [x] If you added a required option, also add it to the template creation (`.github/create_templates.ts`) _not necessary_
-   [x] Add your changes to `CHANGELOG.md` (referencing this PR or the issue you fixed)

**Description:**  
- all arguments are now documented and shown when typing `--help`
- all arguments have shortcuts
- arguments can be passed using environment variables (mainly for testing)
- unknown arguments are no longer allowed

### CODE REVIEW

**Comment by AlCalzone** [comment]
>     * arguments can be passed using environment variables (mainly for testing)

How does that work? Could you add an example to the README please?

**Comment by AlCalzone** [comment]
>     * all arguments have shortcuts

And how would that look like? Are they used with `=` like the long versions or a space before the argument?

**Review by AlCalzone — CHANGES_REQUESTED** [review]
See comments above

**Comment by UncleSamSwiss** [comment]
> > ```
> > * arguments can be passed using environment variables (mainly for testing)
> > ```
> 
> How does that work? Could you add an example to the README please?

I will add this to the documentation. Just for reference, the environment variable feature works as follows:
https://yargs.js.org/docs/#api-reference-envprefix

**Comment by UncleSamSwiss** [comment]
> > ```
> > * all arguments have shortcuts
> > ```
> 
> And how would that look like? Are they used with `=` like the long versions or a space before the argument?

As far as I know, yargs always supports `--parameter=abc` and `--parameter abc` for parameters with a value.
This change adds the possibility to say `-p=abc` or `-p abc` (notice the single dash).

**Comment by UncleSamSwiss** [comment]
Note: failing build because of a 502 error - not related to my changes.

File: README.md [hunk 1]
```diff
@@ -37,8 +37,10 @@ After a short while, you will be asked a few questions. Afterwards all the neces
 
 The following CLI options are available:
 
--   `--target=/path/to/dir` - Specify which directory the adapter files should be created in (instead of the current dir)
--   `--skipAdapterExistenceCheck` - Don't check if an adapter with the same name already exists on `npm`.
+-   `--target=/path/to/dir` - Specify which directory the adapter files should be created in (instead of the current dir). Shortcut: `-t`
+-   `--skipAdapterExistenceCheck` - Don't check if an adapter with the same name already exists on `npm`. Shortcut: `-x`
+
+All CLI options can also be [provided as environment varialbes](https://yargs.js.org/docs/#api-reference-envprefix) by prepending `CREATE_ADAPTER_`. Example: `CREATE_ADAPTER_TARGET=/tmp/iobroker/create-adapter/`
```
Comment (AlCalzone) [inline README.md]:
```diff
- varialbes
+ variables
```

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Whoopsie 🤷🏻‍♂️

File: README.md (hunk 1 above)
Comment (UncleSamSwiss) [reply README.md]: Fixed.

**Review by AlCalzone — APPROVED** [review]

**Comment by AlCalzone** [comment]
merged in 777f2cfee9c5577555eb0ee8437de47dd3a9020f

**Pull request closed without merging**